waspc/src/Generator/ServerGenerator.hs

```diff
@@ -63,12 +64,13 @@ genServer wasp _ =
 --   for progress of this.
 preCleanup :: Wasp -> Path Abs (Dir ProjectRootDir) -> CompileOptions -> IO ()
 preCleanup _ outDir _ = do
-  -- If .env gets removed but there is old .env file in generated project from previous attempts,
-  -- we need to make sure we remove it.
-  removeFile dotEnvAbsFilePath
-    `catch` \e -> when (not $ isDoesNotExistError e) $ throwIO e
-  where
-    dotEnvAbsFilePath = SP.toFilePath $ outDir </> C.serverRootDirInProjectRootDir </> dotEnvInServerRootDir
+  -- TODO: this is already defined in Command/Db/Migrate.hs and should be moved into Lib, issue actually tackled by #105
+  let dbMigrationsDirInDbRootDir = SP.fromPathRelDir [P.reldir|migrations|]
+  let dbMigrationsDirInGenProjectDirAbs = SP.toFilePath $ outDir </> dbRootDirInProjectRootDir </> dbMigrationsDirInDbRootDir
+  let dotEnvAbsFilePath = SP.toFilePath $ outDir </> C.serverRootDirInProjectRootDir </> dotEnvInServerRootDir
+
+  removePathForcibly dbMigrationsDirInGenProjectDirAbs `catch` \e -> when (not $ isDoesNotExistError e) $ throwIO e
```

Comment:
So this works now, with `removePathForcibly`? Any idea why it wasn't working before?